aeppl/transforms.py

```diff
@@ -205,6 +213,180 @@ def apply(self, fgraph: FunctionGraph):
+    # the respective measurable input.
+    backward_value = op.transform_elemwise.backward(value, *other_inputs)
+    input_logprob = logprob(measurable_input, backward_value, **kwargs)
+
+    jacobian = op.transform_elemwise.log_jac_det(value, *other_inputs)
+
+    return input_logprob + jacobian
+
+
+@local_optimizer([Elemwise])
+def find_measurable_transforms(
+    fgraph: FunctionGraph, node: Node
+) -> Optional[List[Node]]:
+    """Find measurable transformations from Elemwise operators."""
+    scalar_op = node.op.scalar_op
+    if not isinstance(scalar_op, (Exp, Log, Add, Mul)):
+        return None
+
+    # Node was already converted
+    if isinstance(node.op, MeasurableVariable):
+        return None  # pragma: no cover
+
+    rv_map_feature: PreserveRVMappings = getattr(fgraph, "preserve_rv_mappings", None)
+    if rv_map_feature is None:
+        return None  # pragma: no cover
+
+    # Check that we have a single source of measurement
+    measurable_inputs = [
+        inp
+        for idx, inp in enumerate(node.inputs)
+        if inp.owner
+        and isinstance(inp.owner.op, MeasurableVariable)
+        and inp not in rv_map_feature.rv_values
+    ]
+
+    if len(measurable_inputs) != 1:
+        return None
+
+    measurable_input: TensorVariable = measurable_inputs[0]
+
+    # Do not apply rewrite to discrete variables
+    if measurable_input.type.dtype.startswith("int"):
+        return None
+
+    # Check that other inputs are not potentially measurable, in which case this rewrite
+    # would be invalid
+    other_inputs = tuple(inp for inp in node.inputs if inp is not measurable_input)
+    if any(
+        ancestor_node
+        for ancestor_node in walk_model(
+            other_inputs,
+            walk_past_rvs=False,
+            stop_at_vars=set(rv_map_feature.rv_values),
+        )
+        if (
+            ancestor_node.owner
+            and isinstance(ancestor_node.owner.op, MeasurableVariable)
+            and ancestor_node not in rv_map_feature.rv_values
+        )
+    ):
+        return None
+
+    # Make base_measure outputs unmeasurable
+    # This seems to be the only thing preventing nested rewrites from being erased
+    measurable_input = assign_custom_measurable_outputs(measurable_input.owner)
+
+    measurable_input_idx = 0
+    transform_inputs: Tuple[TensorVariable, ...] = (measurable_input,)
+    transform: RVTransform
+    if isinstance(scalar_op, Exp):
+        transform = ExpTransform()
+    elif isinstance(scalar_op, Log):
+        transform = LogTransform()
+    elif isinstance(scalar_op, Add):
+        transform_inputs = (measurable_input, at.add(*other_inputs))
+        transform = LocTransform(
+            transform_args_fn=lambda *inputs: inputs[-1],
+        )
+    else:
+        transform_inputs = (measurable_input, at.mul(*other_inputs))
+        transform = ScaleTransform(
+            transform_args_fn=lambda *inputs: inputs[-1],
+        )
```

Comment:
This seems like very dispatch-worthy logic.

Reply:
Yeah, I tried to make it more agnostic than what is actually needed for the current rewrites but I am still not sure what information will be needed going forward. Now I realize that for non-commutative Ops we will also need to know which of the inputs is measurable. There might be other details I am missing. Shall we leave it as is reassess when we expand this with more transforms?

Reply:
Yeah, it's fine as is for now.